### DISCUSSION TITLE
GSheet.ready() doesn't work

### DISCUSSION THREAD

**Discussion by PedroHenriqueCavalcante** [mobizt/ESP-Google-Sheet-Client#33] [Q&A] [answered]
Hi. My code doesn't pass to "if (GSheet.ready() == true){}". Is there any problem with the library? Help me!

`
void loop() {

Serial.println ("Passou_2");

ready = GSheet.ready(); //ready is a boolean variable

if (ready == true){ 

Serial.println ("Passou_3");
`

"Passou_3" is doesn't printed on serial monitor. Consequently, the data doesn't appear in the spreadsheet

**Comment by mobizt** [comment]
Which device you use?

The function `tokenStatusCallback` should print the authentication status.

**Reply by PedroHenriqueCavalcante** [reply]
I'm using ESP32 WROOM-U.

**Reply by mobizt** [reply]
ESP32 WROOM-32U?

The ESP32 variants are.
ESP32.
ESP32-S2.
ESP32-S3.
ESP32-C2.
ESP32-C3.
ESP32-C6.
ESP32-H2.

ESP32 should work with this library normally. 

You should test with the example code first and it should work otherwise you have problem with your device.

**Reply by PedroHenriqueCavalcante** [reply]
Error:

11:49:21.336 -> Token info: type = OAuth2.0 access token, status = on exchange request
11:49:21.962 -> Token info: type = OAuth2.0 access token, status = error
11:49:21.962 -> Token error: code: -1, message: Missing required parameter: assertion

Do you know the solucion?

**Reply by mobizt** [reply, marked as answer]
This is because you change the time after `GSheet.begin`.

You already know that the authentication method relies on time please don't use any method to change your device time after it begins.

Otherwise, you have to call `GSheet.begin` again to re-authenticate.
